### PR TITLE
Adding block moving to top and to bottom functionality.

### LINKED ISSUE #1191: Provide shortcuts to move block to top/bottom of post
## Problem

Moving a block to the top or bottom of a post is cumbersome.

## Proposal

Eventually, we will have true drag and drop reordering and Layout mode. However until that is in place, I'd like to consider adding shortcuts to move a selected block to the top or bottom of the post/page.

This could be as simple as an ActionSheet/BottomSheet that shows options for `Move to Top` and `Move to Bottom` when the user long-presses on the `up`/`down` arrows on the inline toolbar, but I'm open to other ideas.

Example:

098AC29C-E0B3-4C96-9386-3CA03C2E7D2A

## Long-term

In the long-term, we will probably build native on-canvas drag-and-drop and/or a special "Layout Mode" that acts similarly to native list/table view selection modes.

With the introduction of drag-and-drop, we might very well get rid of the up/down arrows on the Inline Toolbar (attached to bottom of block).

### PR DESCRIPTION
**Pull request by illusaen**
Fixes #1191 
`Gutenberg link:` [PR 27554](https://github.com/WordPress/gutenberg/pull/27554)

#### Case 1
1. Insert paragraph block. Type text.
2. Insert second paragraph block. Type text.
3. Insert header block.
4. Long press up button on header block.
5. Press "Move to top" on action sheet that pops up.
6. Verify that header block is on top.

#### Case 2
1. Insert paragraph block. Type text.
2. Insert second paragraph block. Type text.
3. Insert header block.
4. Select first paragraph block.
4. Long press down button on paragraph block.
5. Press "Move to bottom" on action sheet that pops up.
6. Verify that the first paragraph block is on bottom.

#### Case 3
1. Insert paragraph block. Type text.
2. Insert second paragraph block. Type text.
3. Insert header block.
4. Long press up button on header block.
5. Press "Cancel" on action sheet that pops up.
6. Verify that no changes have been made in block order.

#### Case 1
1. Insert paragraph block. Type text.
2. Insert second paragraph block. Type text.
3. Insert header block.
4. Select first paragraph block.
4. Long press down button on paragraph block.
5. Press "Cancel" on action sheet that pops up.
6. Verify that no changes have been made in block order.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/docs/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Comment by hypest** [comment]
👋 @illusaen , thanks for the PR!

A couple of suggestions if I may:
1. What do you think about using the "Fixes #<issuenumber>" format in the beginning of the PR description, so the ticket gets automatically closed when this PR merges? See this [relevant GitHub documentation](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
2. Let's add a "To test:" section, to have exact steps to follow to test the functionality. I see there's a "How it was tested" section in the Gutenberg PR, which you can reshape into a list of steps for the tester/reviewer to follow. For example, something like the "Testing" section [in this PR](https://github.com/WordPress/gutenberg/pull/25031). Thanks!

**Comment by illusaen** [comment]
> 👋 @illusaen , thanks for the PR!
> 
> A couple of suggestions if I may:
> 
>     1. What do you think about using the "Fixes #" format in the beginning of the PR description, so the ticket gets automatically closed when this PR merges? See this [relevant GitHub documentation](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Yes that makes sense!

>     2. Let's add a "To test:" section, to have exact steps to follow to test the functionality. I see there's a "How it was tested" section in the Gutenberg PR, which you can reshape into a list of steps for the tester/reviewer to follow. For example, something like the "Testing" section [in this PR](https://github.com/WordPress/gutenberg/pull/25031). Thanks!

Quick question on that -- I added that section for the gutenberg PR -- should I copy/paste it here?

**Comment by cameronvoell** [comment]
Hi @illusaen , when I updated the CI on this PR to run the full suite of Android and iOS e2e tests, it looks like the new Block Mover Tests are failing on Android, and we are seeing some failures on iOS as well. I was able to confirm the Android e2e test failures locally. 

Can you confirm whether you were able to see the e2e tests all passing locally for both Android and iOS?

**Comment by hypest** [comment]
> Quick question on that -- I added that section for the gutenberg PR -- should I copy/paste it here?

I noticed the list there, but I'd recommend an expanded one (say, a bulleted list for each test case) with simple steps on each line to follow easily. I understand the current format is an abbreviated one with one test per bullet, right?

By the way, a nit: Looks like there's a space character inside the `[ x]` marks in the PR description, and GitHub markdown fails to recognize that as a checked checkmark. Not sure if you intentionally have that like that?

**Comment by illusaen** [comment]
> I noticed the list there, but I'd recommend an expanded one (say, a bulleted list for each test case) with simple steps on each line to follow easily. I understand the current format is an abbreviated one with one test per bullet, right?

Makes sense, changed!

> By the way, a nit: Looks like there's a space character inside the `[ x]` marks in the PR description, and GitHub markdown fails to recognize that as a checked checkmark. Not sure if you intentionally have that like that?

So I didn't actually realize that that was a specific markdown thing; fixed!

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "gutenberg"]
 	path = gutenberg
-	url = ../../WordPress/gutenberg.git
+	url = ../../illusaen/gutenberg.git
```
Comment (cameronvoell) [inline .gitmodules]: Temporary for testing CI, will revert this line before merging

**Review by cameronvoell — APPROVED** [review]
Changes look good. Full tests passed [here](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/9676/workflows/60e98403-2a8c-4db3-9c32-c95ac05b7a32) . Nice work @illusaen !

**Pull request merged**